### PR TITLE
implement support for funds data

### PR DESCRIPTION
**Pull request by ericpien**
Changes:
- adding to scraper a new class `FundsData`.
- Ticker can now reference the data via `ticker.funds_data` when appropriate. When the ticker symbol is not that of a fund's, the code will throw an error when a parameter of `funds_data` is called. i.e. `yf.Ticker('AAPL').funds_data.description`
- Add unit tests in `test_ticker.py` for the funds data
- minor change to `test_ticker.py` as `YFChartError` was removed from code in https://github.com/ranaroussi/yfinance/pull/2000

Resolves:
- https://github.com/ranaroussi/yfinance/issues/244
- https://github.com/ranaroussi/yfinance/issues/422
- https://github.com/ranaroussi/yfinance/issues/684
- https://github.com/ranaroussi/yfinance/discussions/1761
- https://github.com/ranaroussi/yfinance/discussions/2012

Sample Code:
``` python
import yfinance as yf
spy_ticker = yf.Ticker('SPY')
spy_funds_data = spy_ticker.funds_data

spy_funds_data.description
spy_funds_data.fund_overview
spy_funds_data.fund_operations
spy_funds_data.asset_classes
spy_funds_data.top_holdings
spy_funds_data.equity_holdings
spy_funds_data.bond_holdings
spy_funds_data.bond_ratings
spy_funds_data.sector_weightings
```

Credits to https://github.com/ranaroussi/yfinance/pull/1784 for the idea

### CODE REVIEW

**Comment by ValueRaider** [comment]
Update readme

**Comment by ericpien** [comment]
> Update readme

Done. Thanks for reviewing.

**Comment by ValueRaider** [comment]
Set df indexes. And some might be better as simple dict.

**Comment by ericpien** [comment]
> Set df indexes. And some might be better as simple dict.

I converted data with just 2 columns to dicts and kept the others as DF with the indices set.

**Comment by ericpien** [comment]
hey @ValueRaider not sure if you've been busy or just missed it so pinging again. thanks.

**Comment by ValueRaider** [comment]
Looks good for `SPY`. I'm just thinking about the exception experience.

I would rephrase this:
> raise YFDataException("Failed to parse quote type. No Fund data found.")
to
> raise YFDataException("No Fund data found.")

Then with:
```
        except IndexError:
            raise YFDataException("Failed to parse fund json data.")
```
don't discard the original exception. See how `base.py` handles exceptions.

**Comment by ericpien** [comment]
Thanks for reviewing @ValueRaider. I interpreted the part about "don't discard the original exception" as meaning to log the base exception so implemented as such.
